javalang/parser.py

```diff
@@ -854,6 +854,15 @@ def parse_field_declarators_rest(self):
         return tree.FieldDeclaration(declarators=declarators)
 
     @parse_debug
+    def parse_lambda_method_body(self):
+        body = None
+        if self.try_accept('->'):
```

Comment:
This conditional should be removed and `self.accept('->')` used instead. It will raise a `JavaSyntaxError` if the `->` is not present. We want to ensure `parse_*` methods fail by raising exceptions if they can not parse what they were requested to parse. As this is, `(String foo, Long bar)` would be accepted by `parse_lambda_expression` and return a `LambdaExpression` instance with a body of `None`.